velox/functions/sparksql/MightContain.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/MightContain.h"
+
+#include "velox/common/base/BloomFilter.h"
+#include "velox/common/memory/HashStringAllocator.h"
+#include "velox/expression/DecodedArgs.h"
+#include "velox/vector/FlatVector.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+class BloomFilterMightContainFunction final : public exec::VectorFunction {
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args, // Not using const ref so we can reuse args
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& resultRef) const final {
+    VELOX_CHECK_EQ(args.size(), 2);
+    context.ensureWritable(rows, BOOLEAN(), resultRef);
+    auto& result = *resultRef->as<FlatVector<bool>>();
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    auto serialized = decodedArgs.at(0);
+    auto value = decodedArgs.at(1);
+
+    HashStringAllocator allocator{context.pool()};
+    VELOX_USER_CHECK(serialized->isConstantMapping())
+    BloomFilter output{StlAllocator<uint64_t>(&allocator)};
+    output.merge(serialized->valueAt<StringView>(0).str().c_str());
```

Comment:
Use `serialized->valueAt<StringView>(0).data()`.

Can output.merge throw? If that's the case, you need to catch the exception and report it via context.setErrors. Otherwise, this function won't work properly under TRY. See https://facebookincubator.github.io/velox/develop/scalar-functions.html#try-expression-support

Reply:
inline const char *facebook::velox::StringView::data() &&
+1 overload

function "facebook::velox::StringView::data() &&" (declared at line 114 of "/mnt/DP_disk1/code/velox/velox/type/StringView.h") cannot be referenced -- it is a deleted function

So I cannot use `serialized->valueAt<StringView>(0).data()`

Reply:
This is actually not safe, so the compiler error is legit - and by design. 

`serialized->valueAt<StringView>(0)` will generate a temporary StringView, so your .data() would return a pointer into the temporary StringView, which gets destructed right away. If you want to do something like this, you need to explicitly save it into the stack to ensure its lifetime:

```
auto sv = serialized->valueAt<StringView>(0);
auto ptr = sv.data();  // all good
```

Reply:
@pedroerp Pedro, thank you for clarifying. Let me send a PR to do that. This is quite inconvenient though and makes developers go with a solution that copies the string.

Reply:
> This is quite inconvenient though and makes developers go with a solution that copies the string.

Agreed it's inconvenient, but the alternative is removing the compiler check at the cost of correctness. Also, the example I send above doesn't copy the string. The compiler should translate into the exact same code, except safer.